src/model/test/Model_GTest.cpp

```diff
@@ -790,3 +792,57 @@ TEST_F(ModelFixture, Ensure_Name_Unicity_SpaceAndSpaceGroupNames) {
     EXPECT_NE(mos[i1].nameString(), mos[i2].nameString());
   }
 }
+
+TEST_F(ModelFixture, Issue_4372) {
+
+  // Open OffsetTests
+  // INtersect surfaces
+  // Match Surfaces
+  // Check the internal walls are paired
+  // Surface 4 Space 101 with SPace 102 surface 12
+  // Surface 2 Space 102 with
+
+  openstudio::path modelPath = resourcesPath() / "model" / toPath("offset_tests.osm");
+  ASSERT_TRUE(openstudio::filesystem::exists(modelPath));
+
+  openstudio::osversion::VersionTranslator vt;
+  boost::optional<openstudio::model::Model> model = vt.loadModel(modelPath);
+  ASSERT_TRUE(model);
+
+  std::vector<Space> spaces = model->getModelObjects<Space>();
+  intersectSurfaces(spaces);
+
+  matchSurfaces(spaces);
+  std::vector<Surface> surfacesAfter = model->getModelObjects<Surface>();
```

Comment:
```suggestion
  std::vector<Surface> surfacesAfter = model->getConcreteModelObjects<Surface>();
```

Reply:
What is the difference between the two calls? Should I always use getConcreteModelObjects or are the occasions where getModelObjects is better?